core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -764,7 +766,7 @@ class Partition(val topicPartition: TopicPartition,
     } else {
       def logEndOffsetString(r: Replica) = s"replica ${r.brokerId}: ${r.logEndOffsetMetadata}"
       trace(s"Skipping update high watermark since new hw $newHighWatermark is not larger than old hw $oldHighWatermark. " +
-        s"All current LEOs are ${assignedReplicas.map(logEndOffsetString)}")
+        s"All current LEOs are ${remoteReplicas.map(logEndOffsetString)}")
```

Comment:
This is another case where it is probably useful to include the LEO of the leader.

Reply:
Done.